torchao/prototype/autoround/core.py

```diff
@@ -0,0 +1,231 @@
+import os
+from dataclasses import dataclass
+from typing import Dict, Tuple
+
+import torch
+from torch.utils._pytree import tree_flatten, tree_unflatten
+
+import torchao.prototype.autoround.utils as ar_utils
+import torchao.quantization as ao_quant
+from torchao.dtypes import TensorCoreTiledLayoutType, to_affine_quantized_static
+from torchao.prototype.autoround.multi_tensor import MultiTensor
+from torchao.quantization.quant_primitives import ZeroPointDomain
+from torchao.utils import find_multiple
+
+# TODO: remove it before merge
+ar_utils.freeze_random()
+
+
+@dataclass
+class AutoRoundConfig:
+    bits: int = 4
+    sym: bool = False
+    iters: int = 200
+    group_size: int = 128
+    train_bs: int = 4
+    eval_bs: int = 4
+    seed: int = 42
+    amp: bool = True
+    nsamples: int = 128
+    seqlen: int = 2048
+    quant_lm_head: bool = False
+
+
+auto_round_config = AutoRoundConfig()
```

Comment:
does this have to be global? or can we just pass this around

Reply:
This is not necessary, I plan to refactor the UX to align with the API outlined in https://github.com/pytorch/ao/pull/721 :)